src/js/components/deployments/deployment-wizard/phasesettings.js

```diff
@@ -0,0 +1,198 @@
+
+    // set value based on unit 
+    value = newPhases[index].delayUnit === 'days' ? value*24 : value;
+    newPhases[index].delay = value;
+
+    this.props.updatePhaseStarts(newPhases);
+    // logic for updating time stamps should be in parent - only change delays here
+  }
+
+  updateBatchSize(value, index) {
+    // When phase's batch size changes, adjust other phases' batch sizes starting with 'remainder' first.
+    let newPhases = this.props.phases;
+
+    if (value < 1) {
+      value = 1;
+    } else if (value >100) {
+      value = 100;
+    }
+
+    newPhases[index].batch_size = value;
+    const remainder = this.getRemainder(newPhases);
+    // if new remainder will be 0 or negative remove last phase
+    if (remainder < 1) {
+      newPhases.pop();
+      newPhases[newPhases.length-1].batch_size = this.getRemainder(newPhases);
+    }
+
+    this.props.deploymentSettings(newPhases, 'phases');
+  }
+
+  addPhase() {
+    let phases = this.props.phases;
+    let newPhase = {};
+
+    // assign new batch size to *previous* last batch
+    const remainder = this.getRemainder(phases);
+    // make it default 10, unless remainder is <=10 in which case make it half remainder
+    let batch_size = (remainder>10) ? 10 : (Math.floor(remainder/2));
+    phases[phases.length-1].batch_size = batch_size;
+
+    // check for previous phase delay or set 2hr default
+    const delay = phases[phases.length-1].delay || 2;
+    phases[phases.length-1].delay = delay;
+    const delayUnit = phases[phases.length-1].delayUnit || 'hours';
+    phases[phases.length-1].delayUnit = delayUnit;
+
+    phases.push(newPhase);
+    // use function to set new phases incl start time of new phase
+    this.props.updatePhaseStarts(phases);
+  }
+
+  removePhase(index) {
+    let phases = this.props.phases;
+    phases.splice(index, 1);
+    if (phases.length===1) {
+      delete phases[0].batch_size;
+      delete phases[0].delay;
+      this.props.deploymentSettings(phases, 'phases');
+    } else {
+      this.props.updatePhaseStarts(phases);
+    }
+  }
+
+  handleDelayToggle(value, index) {
+    let phases = this.props.phases;
+
+    if (phases[index].delayUnit === 'days' && value === 'hours') {
+      phases[index].delay = Math.floor(phases[index].delay/24);
+    }
+    if (phases[index].delayUnit !== 'days' && value === 'days')  {
+      phases[index].delay =  phases[index].delay*24;
+    }
+
+    phases[index].delayUnit = value;
+    this.props.updatePhaseStarts(phases);
+  }
+
+  render() {
+    const self = this;
+    const props = self.props;
+    const remainder = self.getRemainder(props.phases);
+
+    // disable 'add phase' if no more devices left
+    const disableAdd = ((remainder/100)*props.numberDevices ) < 1;
+  
+    const phases = props.phases ? props.phases.map((phase, index) => {
+      let max = index > 0 ? 100-(props.phases[index-1].batch_size) : 100;
+      const deviceCount = Math.ceil((props.numberDevices / 100) * phase.batch_size);
```

Comment:
for the last phase the `batch_size` might not be set so the fallback should be `... * (phase.batch_size || remainder)`